kotlin/android/app/src/main/java/dev/firezone/android/features/session/ui/SessionActivity.kt

```diff
@@ -34,19 +54,30 @@ internal class SessionActivity : AppCompatActivity() {
         binding = ActivitySessionBinding.inflate(layoutInflater)
         setContentView(binding.root)
 
+        // Bind to existing TunnelService
+        val intent = Intent(this, TunnelService::class.java)
+        bindService(intent, serviceConnection, Context.BIND_AUTO_CREATE)
+
         setupViews()
         setupObservers()
-        viewModel.connect(this@SessionActivity)
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        if (serviceBound) {
+            unbindService(serviceConnection)
+            serviceBound = false
+        }
     }
 
     private fun setupViews() {
         binding.btSignOut.setOnClickListener {
-            viewModel.disconnect()
+            Log.d(TAG, "Sign out button clicked")
+            viewModel.clearToken()
```

Comment:
Does the actor name get cleared here, too?

Reply:
Good catch! I believe it does not, I'll fix that now.